msgpack_unpack.c

```diff
@@ -694,66 +556,70 @@ int msgpack_unserialize_map_item(
         }
     }
 
-    if (Z_TYPE_PP(container) != IS_ARRAY && Z_TYPE_PP(container) != IS_OBJECT)
-    {
+    zend_bool container_is_ref = 0;
```

Comment:
@laruence 
What is the proper way to ZVAL_DEREF *container here, without actually modifying *container?

Calling array_init on the result of Z_REFVAL_P causes issues, I just want to modify the REFVAL of *container, without harming the ref *container